pygdf/dataframe.py

```diff
@@ -594,6 +594,144 @@ def _n_largest_or_smallest(self, method, n, columns, keep):
                 df[k] = self[k].reset_index().take(new_positions)
         return df.set_index(self.index.take(new_positions))
 
+    def merge(self, other, on=None, how='multi-left', lsuffix='_x', rsuffix='_y'):
+        if how != 'multi-left':
+            raise ValueError('{!r} join not implemented yet'.format(how))
+
+        same_names = set(self.columns) & set(other.columns)
+        if same_names and not (lsuffix or rsuffix):
+            raise ValueError('there are overlapping columns but '
+                             'lsuffix and rsuffix are not defined')
+
+        lhs = self
+        rhs = other
+        # XXX: Replace this stub
+        #joined_values, joined_indicies = self._stub_merge(
+        #    lhs, rhs, left_on=on, right_on=on, how=how, return_joined_indicies=True
+        #    )
+        joined_values, joined_indicies = self._merge_gdf(
```

Comment:
`_merge_gdf` can't handle empty dataframe. It leads to a `std::bad_alloc` in libgdf.  I am fixing it in my branch with https://github.com/sklam/pygdf/commit/6f2f57469d2201bb3ae0a9f93a3c1411ec15b065, which is needed for https://github.com/gpuopenanalytics/dask_gdf/pull/22.